src/commonTest/kotlin/com/github/quillraven/fleks/collection/ListMutableEntityBagTest.kt

```diff
@@ -0,0 +1,574 @@
+package com.github.quillraven.fleks.collection
+
+import com.github.quillraven.fleks.Entity
+import com.github.quillraven.fleks.configureWorld
+import kotlin.test.*
+
+class ListMutableEntityBagTest {
+
+    private val testEntity1 = Entity(0, version = 0u)
+    private val testEntity2 = Entity(1, version = 0u)
+    private val testBag = ListMutableEntityBag(ArrayList(2)).apply {
+        this += testEntity1
+        this += testEntity2
+    }
+
+    private fun bagOf(entity: Entity) = ListMutableEntityBag(ArrayList(1)).apply {
+        this += entity
+    }
+
+    private fun bagOf(entity1: Entity, entity2: Entity) = ListMutableEntityBag(ArrayList(2)).apply {
+        this += entity1
+        this += entity2
+    }
+
+    private fun arrayBagOf(entity: Entity) = ArrayMutableEntityBag(1).apply {
```

Comment:
Many functions from the ListMutableEntityBag return an ArrayMutableEntityBag, which is intentional, so I have adapted the tests for it

Reply:
With the MutableList it is impossible to get the capacity variable so I have removed the tests.